### DISCUSSION TITLE
adding a 5D version of reduce

### DISCUSSION THREAD

**Discussion by francois-rincon** [idefix-code/idefix#263] [General]
hello, I had to add a 5D version of idefix_reduce on my patou fork of idefix, just as we did last year for the idefix_for loops.
I think that this could clash with idefix when I try to merge in the future (or break compatibility), so I suggest that the loop be absorbed into idefix itself as it is a very benign piece of code.

// 5D loop
template <typename Function, typename Reducer>
inline void idefix_reduce(const std::string & NAME,
                       const int NB, const int NE,
                       const int NPRIMEB, const int NPRIMEE,
                       const int KB, const int KE,
                       const int JB, const int JE,
                       const int IB, const int IE,
                       Function function,
                       Reducer redFunction) {
    // We only implement MDRange reductions here since the other implementations are too
    // complicated to be implemented for any reduction operator on any class
    #ifdef DEBUG
    idfx::pushRegion("idefix_reduce("+NAME+")");
    #endif
    Kokkos::parallel_reduce(NAME,
        Kokkos::MDRangePolicy<Kokkos::Rank<5,Kokkos::Iterate::Right, Kokkos::Iterate::Right>>
          ({NB,NPRIMEB,KB,JB,IB},{NE,NPRIMEE,KE,JE,IE}), function, redFunction);

    #ifdef DEBUG
    Kokkos::fence();
    idfx::popRegion();
    #endif
}

**Comment by francois-rincon** [comment]
(it does compile, I think it works too but who knows :) )

**Comment by glesur** [comment]
YEs, this can (should!) be added to Idefix as well. Feel free to propose a PR with the modifications in` loop.hpp`.
